src/utils/configBuilders.ts

```diff
@@ -157,31 +157,28 @@ export const buildRollupConfigPayload = ({
 }: {
   rollupConfig: RollupConfig;
   chainConfig: string;
-}): GetFunctionArgs<typeof rollupCreatorABI, 'createRollup'>['args'][0] => {
+}): GetFunctionArgs<typeof rollupCreatorABI, 'createRollup'>['args']['0']['config'] => {
   try {
-    const rollupConfigPayload: RollupConfigPayload = {
-      ...rollupConfig,
-      chainConfig,
-      baseStake: parseEther(String(rollupConfig.baseStake)),
-    };
-
-    assertIsAddress(rollupConfigPayload.owner);
-    assertIsAddress(rollupConfigPayload.stakeToken);
+    assertIsAddress(rollupConfig.owner);
+    assertIsAddress(rollupConfig.stakeToken);
 
     return {
-      ...rollupConfigPayload,
```

Comment:
Removed the spreading so it's more explicit and readable what this object contains